search-app/src/cmr/search/services/query_execution/facets/hierarchical_v2_facets.clj

```diff
@@ -178,22 +200,24 @@
   elastic-aggregations - the portion of the elastic aggregations response to parse to generate
                          the part of the facets v2 response related to the passed in field."
   [recursive-parse-fn has-siblings-fn generate-links-fn field field-hierarchy elastic-aggregations]
-  ;; Each value for this field has its own bucket in the elastic aggregations response
-  ;; Top level could be a field
-  (let [buckets (or (get-in elastic-aggregations [field :buckets])
-                    (get elastic-aggregations :buckets))]
-    (for [bucket buckets
-          :let [value (extract-value-from-bucket bucket field)
-                count (get-in bucket [:coll-count :doc_count] (:doc_count bucket))
-                sub-facets (recursive-parse-fn value bucket)
-                ;; Sort alphabetically
-                sub-facets (when (seq (:children sub-facets))
-                             (update sub-facets :children
-                                     #(sort-by :title util/compare-natural-strings %)))
-                children-values-to-remove (find-applied-children sub-facets field-hierarchy false)
-                has-siblings? (has-siblings-fn value)
-                links (generate-links-fn value has-siblings? children-values-to-remove)]]
-      (v2h/generate-hierarchical-filter-node value count links sub-facets))))
+  (let [temporal-case? (some? (get elastic-aggregations :buckets))]
+    (if (and temporal-case? (not= 1 (count field-hierarchy)))
+      (generate-temporal-hierarchical-children recursive-parse-fn has-siblings-fn generate-links-fn
+                                               field field-hierarchy elastic-aggregations)
+      (let [buckets (or (get-in elastic-aggregations [field :buckets])
+                        (get elastic-aggregations :buckets))]
+        (for [bucket buckets
```

Comment:
What I typically do in situations like this split out the logic that operates on a single element into its own function, then use `map ` on that function. Separating how we work with a single item and how we work with a collection of items is almost always a good idea.

Reply:
Refactored.